rust/src/http2/http2.rs

```diff
@@ -0,0 +1,1175 @@
+    let event = match c_event_name.to_str() {
+        Ok(s) => {
+            match s {
+                "invalid_frame_header" => HTTP2Event::InvalidFrameHeader as i32,
+                "invalid_client_magic" => HTTP2Event::InvalidClientMagic as i32,
+                "invalid_frame_data" => HTTP2Event::InvalidFrameData as i32,
+                "invalid_header" => HTTP2Event::InvalidHeader as i32,
+                "invalid_frame_length" => HTTP2Event::InvalidFrameLength as i32,
+                "extra_header_data" => HTTP2Event::ExtraHeaderData as i32,
+                "long_frame_data" => HTTP2Event::LongFrameData as i32,
+                "stream_id_reuse" => HTTP2Event::StreamIdReuse as i32,
+                _ => -1, // unknown event
+            }
+        }
+        Err(_) => -1, // UTF-8 conversion failed
+    };
+    unsafe {
+        *event_type = core::APP_LAYER_EVENT_TYPE_TRANSACTION;
+        *event_id = event as std::os::raw::c_int;
+    };
+    0
+}
+
+#[no_mangle]
+pub extern "C" fn rs_http2_state_get_event_info_by_id(
+    event_id: std::os::raw::c_int, event_name: *mut *const std::os::raw::c_char,
+    event_type: *mut core::AppLayerEventType,
+) -> i8 {
+    if let Some(e) = HTTP2Event::from_i32(event_id as i32) {
+        let estr = match e {
+            HTTP2Event::InvalidFrameHeader => "invalid_frame_header\0",
+            HTTP2Event::InvalidClientMagic => "invalid_client_magic\0",
+            HTTP2Event::InvalidFrameData => "invalid_frame_data\0",
+            HTTP2Event::InvalidHeader => "invalid_header\0",
+            HTTP2Event::InvalidFrameLength => "invalid_frame_length\0",
+            HTTP2Event::ExtraHeaderData => "extra_header_data\0",
+            HTTP2Event::LongFrameData => "long_frame_data\0",
+            HTTP2Event::StreamIdReuse => "stream_id_reuse\0",
+        };
+        unsafe {
+            *event_name = estr.as_ptr() as *const std::os::raw::c_char;
+            *event_type = core::APP_LAYER_EVENT_TYPE_TRANSACTION;
+        };
+        0
+    } else {
+        -1
+    }
+}
+#[no_mangle]
+pub extern "C" fn rs_http2_state_get_tx_iterator(
+    _ipproto: u8, _alproto: AppProto, state: *mut std::os::raw::c_void, min_tx_id: u64,
+    _max_tx_id: u64, istate: &mut u64,
+) -> applayer::AppLayerGetTxIterTuple {
+    let state = cast_pointer!(state, HTTP2State);
+    match state.tx_iterator(min_tx_id, istate) {
+        Some((tx, out_tx_id, has_next)) => {
+            let c_tx = unsafe { transmute(tx) };
+            let ires = applayer::AppLayerGetTxIterTuple::with_values(c_tx, out_tx_id, has_next);
+            return ires;
+        }
+        None => {
+            return applayer::AppLayerGetTxIterTuple::not_found();
+        }
+    }
+}
+
+#[no_mangle]
+pub extern "C" fn rs_http2_getfiles(
+    state: *mut std::os::raw::c_void, direction: u8,
+) -> *mut FileContainer {
+    let state = cast_pointer!(state, HTTP2State);
+    //TODO8 understand inversion ?
```

Comment:
It seems `PrefilterTxFiledata` calls `AppLayerParserGetFiles` with the wrong direction

So, I guess there is an inversion somewhere...